Jellyfin.Networking/Manager/NetworkManager.cs

```diff
@@ -0,0 +1,1283 @@
+        {
+            if (!_eventfire)
+            {
+                _logger.LogDebug("Network Address Change Event.");
+                // As network events tend to fire one after the other only fire once every second.
+                _eventfire = true;
+                _ = OnNetworkChangeAsync();
+            }
+        }
+
+        /// <summary>
+        /// Parses the user defined overrides into the dictionary object.
+        /// Overrides are the equivalent of localised publishedServerUrl, enabling
+        /// different addresses to be advertised over different subnets.
+        /// format is subnet=ipaddress|host|uri
+        /// when subnet = 0.0.0.0, any external address matches.
+        /// </summary>
+        private void InitialiseOverrides(ServerConfiguration config)
+        {
+            lock (_intLock)
+            {
+                _publishedServerUrls.Clear();
+                string[] overrides = config.PublishedServerUriBySubnet;
+                if (overrides == null)
+                {
+                    return;
+                }
+
+                foreach (var entry in overrides)
+                {
+                    var parts = entry.Split('=');
+                    if (parts.Length != 2)
+                    {
+                        _logger.LogError("Unable to parse bind override. {0}", entry);
+                    }
+                    else
+                    {
+                        var replacement = parts[1].Trim();
+                        if (string.Equals(parts[0], "remaining", StringComparison.OrdinalIgnoreCase))
+                        {
+                            _publishedServerUrls[new IPNetAddress(IPAddress.Broadcast)] = replacement;
+                        }
+                        else if (string.Equals(parts[0], "external", StringComparison.OrdinalIgnoreCase))
+                        {
+                            _publishedServerUrls[new IPNetAddress(IPAddress.Any)] = replacement;
+                        }
+                        else if (TryParseInterface(parts[0], out NetCollection? addresses) && addresses != null)
+                        {
+                            foreach (IPNetAddress na in addresses)
+                            {
+                                _publishedServerUrls[na] = replacement;
+                            }
+                        }
+                        else if (IPNetAddress.TryParse(parts[0], out IPNetAddress result))
+                        {
+                            _publishedServerUrls[result] = replacement;
+                        }
+                        else
+                        {
+                            _logger.LogError("Unable to parse bind ip address. {0}", parts[1]);
+                        }
+                    }
+                }
+            }
+        }
+
+        private void InitialiseBind(ServerConfiguration config)
+        {
+            string[] ba = config.LocalNetworkAddresses;
```

Comment:
`ba` is a really bad variable name, no offense.

Reply:
renamed variable.